### PR TITLE
Protobuf struct properly converted to dict, fix compile time error fo…

### PR DESCRIPTION
**Pull request by max-hoffman**
Signed-off-by: Max Hoffman <max@dolthub.com>

# TL;DR

This originally fixed a bug with `DoltConfig` parsing in `to_python_value`, but expanded to a few other changes:
- `dolt` dependency is downloaded in a pytest fixture
- `to_python_value` with bad literal value raises
- added TypeEngine transformer unit tests
- added one more workflow integration test with branching
- plugin folder name is `flytekit-dolt` to avoid pip package naming clash with plugin name `flytekitplugins-dolt`

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [x] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/lyft/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/lyft/flyte/issues/<number>_

### CODE REVIEW

**Review by wild-endeavor — APPROVED** [review]

File: plugins/flytekit-dolt/scripts/install.sh [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+}
+
+assert_linux_or_macos
+assert_dependencies
+create_workdir
+install_binary_release
+
+}
+
+_ "$0" "$@"
```
Comment (wild-endeavor) [inline plugins/flytekit-dolt/scripts/install.sh:112]: what does _ do?

**Review by wild-endeavor — APPROVED** [review]

File: plugins/flytekit-dolt/scripts/install.sh (hunk 1 above)
Comment (wild-endeavor) [reply plugins/flytekit-dolt/scripts/install.sh:112]: nice.
